src/qt/clientmodel.cpp

```diff
@@ -336,3 +337,20 @@ void ClientModel::unsubscribeFromCoreSignals()
     uiInterface.NotifyBlockTip.disconnect(boost::bind(BlockTipChanged, this, _1, _2, false));
     uiInterface.NotifyHeaderTip.disconnect(boost::bind(BlockTipChanged, this, _1, _2, true));
 }
+
+UniValue ClientModel::getProxyInfo() const
```

Comment:
Suggestion to remove `UniValue` dependency:
```cpp
bool ClientModel::GetProxyInfo(std::string& ip_port) const
{
    for (int n = 0; n < NET_MAX; ++n) {
        proxyType proxy;
        if (GetProxy((Network) n, proxy)) {
            ip_port = proxy.proxy.ToStringIPPort();
            return true;
        }
    }
    return false;
}
```
Haven't tested.

Reply:
I can not see how the new icon relates to proxy. Instead of showing "something with computer and cable", maybe simply show a "P" ?